From 2591fcf1c5ef0e650210a24d91c83ba2f1bba3e8 Mon Sep 17 00:00:00 2001
From: Gregory Norton <gregory@serenade.local>
Date: Wed, 16 Sep 2026 17:29:36 -0400
Subject: [PATCH] Services: Build FileSystemAccessServer on host

Apps open and save files through this service (ImageViewer, PixelPaint,
TextEditor), so it must be part of the Lagom build. Its endpoint headers are
already generated by the Lagom root for LibFileSystemAccessClient, so skip
the service-side codegen there, as with the other host-built services.
---
 Userland/Services/CMakeLists.txt                        | 4 ++--
 Userland/Services/FileSystemAccessServer/CMakeLists.txt | 9 +++++++--
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/Userland/Services/CMakeLists.txt b/Userland/Services/CMakeLists.txt
index 2c4104e8ee..ccfc8b3301 100644
--- a/Userland/Services/CMakeLists.txt
+++ b/Userland/Services/CMakeLists.txt
@@ -1,7 +1,9 @@
+add_subdirectory(AudioServer)
 add_subdirectory(Clipboard)
 add_subdirectory(ConfigServer)
 add_subdirectory(EchoServer)
 add_subdirectory(FileOperation)
+add_subdirectory(FileSystemAccessServer)
 add_subdirectory(ImageDecoder)
 add_subdirectory(LaunchServer)
 add_subdirectory(LookupServer)
@@ -12,14 +14,12 @@ add_subdirectory(Taskbar)
 add_subdirectory(WebContent)
 add_subdirectory(WebServer)
 add_subdirectory(WindowServer)
-add_subdirectory(AudioServer)
 
 if (SERENITYOS)
     add_subdirectory(ChessEngine)
     add_subdirectory(CrashDaemon)
     add_subdirectory(DeviceMapper)
     add_subdirectory(DHCPClient)
-    add_subdirectory(FileSystemAccessServer)
     add_subdirectory(KeyboardPreferenceLoader)
     add_subdirectory(LaunchServer)
     add_subdirectory(LoginServer)
diff --git a/Userland/Services/FileSystemAccessServer/CMakeLists.txt b/Userland/Services/FileSystemAccessServer/CMakeLists.txt
index 851a127d6f..bc3a88f99c 100644
--- a/Userland/Services/FileSystemAccessServer/CMakeLists.txt
+++ b/Userland/Services/FileSystemAccessServer/CMakeLists.txt
@@ -4,8 +4,13 @@ serenity_component(
     TARGETS FileSystemAccessServer
 )
 
-compile_ipc(FileSystemAccessServer.ipc FileSystemAccessServerEndpoint.h)
-compile_ipc(FileSystemAccessClient.ipc FileSystemAccessClientEndpoint.h)
+# On host systems the endpoint headers are generated by the Lagom build itself
+# (needed by LibFileSystemAccessClient); generating them here too would create
+# duplicate codegen targets.
+if (SERENITYOS)
+    compile_ipc(FileSystemAccessServer.ipc FileSystemAccessServerEndpoint.h)
+    compile_ipc(FileSystemAccessClient.ipc FileSystemAccessClientEndpoint.h)
+endif()
 
 set(SOURCES
     ConnectionFromClient.cpp
-- 
2.55.0

